tests/test_utils.py

```diff
@@ -1,9 +1,266 @@
+from datetime import datetime, timedelta
+from os import path
+from io import StringIO
+import sys
+from typing import Dict, List, Tuple
+
+from requests import get
+from requests.exceptions import JSONDecodeError
+
+from flask import Response
+import rich
+from rich.table import Table
+
+from _pytest.capture import CaptureFixture
+from requests_mock.adapter import _Matcher
 from requests_mock.mocker import Mocker
 from pytest import raises
+from pyfakefs.fake_filesystem import FakeFilesystem
+from unittest import mock
+from unittest.mock import MagicMock
 
 from dds_cli import DDSEndpoint
-from dds_cli.exceptions import ApiResponseError, DDSCLIException
-from dds_cli.utils import perform_request
+from dds_cli.exceptions import (
+    ApiRequestError,
+    ApiResponseError,
+    DDSCLIException,
+    NoDataError,
+    TokenDeserializationError,
+    TokenExpirationMissingError,
+)
+from dds_cli.utils import (
+    create_table,
+    delete_folder,
+    format_api_response,
+    get_deletion_confirmation,
+    get_json_response,
+    get_required_in_response,
+    get_token_expiration_time,
+    get_token_header_contents,
+    multiple_help_text,
+    perform_request,
+    print_or_page,
+    readable_timedelta,
+    sort_items,
+)
+
+sample_fully_authenticated_token = (
+    "eyJhbGciOiJBMjU2S1ciLCJlbmMiOiJBMjU2R0"
+    "NNIiwiZXhwIjoiMjAyMi0wMi0yNFQxNDo1MTow"
+    "Ni4yMTg3OTgifQ.Ul5rfhy0S9iaX2dPGH93HtL"
+    "-3tVdGBdAzzoTQXb_QJrcIIA0wEwdQw.95ii5p"
+    "anPoIUV1Mf.cAwnDuri4kxjwnQfY48pS0rZ-ob"
+    "-RnKBacUcOe0l3RJrMCbc2nfkdkzc7KBH06ESi"
+    "D-I7MU-U6270uLa2M4ZcLk0AkCZ3S7xrm9-bDu"
+    "_73yCDCIQravwphlxCVSSrNQUPU8BonwBuDu-5"
+    "WjuJyL_zC7MBcduxau8L0Hpk0IOLfIDgEtq9uR"
+    "ELIxjbw1-YEhOtUBKm3E3jevmohgCt7RqcbbuB"
+    "ZtZgYSm5NjOO1XhHBz_kZo1lhONNVVDNUkAoAP"
+    "FoJ7WOAPajCGiDi8yyq7e-ojcxoSf0gl5NVd25"
+    "cmO7i4OqsXB9VNlN5asEZE4WXAmVrQTppCbTG_"
+    "9te04fCDwGabzDqtdfUqX-d_yaQ_UYHmJMN1xc"
+    "4aF-uWZtk3loyMZU-uedQPqsJSZ.ay0MIzbtmt"
+    "GsxUm2blaKUA"
+)
+
+token_without_exp_claim_in_header = (
+    "eyJhbGciOiJBMjU2S1ciLCJlbmMiOiJBMjU2R"
+    "0NNIn0.3H7fZh-rxkSuERSgknz4fOtseDn6PN"
+    "c0RR-1IU8EmoTfpOuMOTvVbg.a_UwR9ArB6kn"
+    "1LEB.7Ko4g1Xs9S_EQsAbGCtc96x_h3P6lwZz"
+    "_X6t1EKA-EFeLXgwjAHuX5S_rC7YK28rqIT_m"
+    "9FQABgTSgi0nBHCUurPA43U2P2mDR9UOvCHFY"
+    "QXLKyO3M-ykVrmNwSGZMjo3HHrmcuICiwiH7l"
+    "boGl5Vr-iSFpyyuy33thSrlwfutI80sKe3RSm"
+    "Kup_Mh7tM0mw0WbQezfAcNR_52BeP_ncbVxFl"
+    "714ikyo2HCk0bKREIpetdaKCaoZgqlhOarlAU"
+    "GwPaKtdgmXb7Ef4VKfYdnLIxqzv3RtVmZiEb1"
+    "L-xCS4vnwXvw_bEa_QU-5HfyLYOszjAHiYHxr"
+    "q8v1xnfoyWfd20OxQMhYueVzlPw1HfMSfvCNV"
+    "LZO-vNNKHTaGnPyGuykhMNScIgkR1l8.TEp4L"
+    "s4c29JtGogmdYbTbw"
+)
```

Comment:
Can you add a comment above these? I'm assuming these are generated with no expiry date?